src/models/gamma_poisson_model.py

```diff
@@ -583,7 +583,7 @@ def by_impressions(
         if len(impressions) != 1:
             raise ValueError("Impressions vector must have a length of 1.")
         hist = self._expected_histogram(
-            impressions[0],
+            min(impressions[0], self._max_impressions - 1),
```

Comment:
I haven't found a particular reason for " - 1".  Could you explain?

Reply:
@matthewclegg appears to be an open question from @jiayu-google. Please resolve.

Reply:
Sorry, I should have explained this earlier.  This was added out of an abundance of caution.  If impressions[0] == self._max_impressions, then the probability that a person with n impressions is shown k of them on behalf of the campaign (see the function _knreach, line 158 and following) becomes:

C(n, k) 1**k 0**(n-k) f(n | alpha, beta),

where C(n, k) is the binomial coefficient counting the number of distinct subsets of size k that can be chosen from a set of size n, and f(n | alpha, beta) is the probability mass function of the Gamma-Poisson distribution with parameters alpha and beta.  In any event, the above expression contains the term 0**0 when n=k, and I was worried that this might cause a floating point exception.  To avoid having to worry about this issue, I clamped impressions[0] at self._max_impressions-1.  For realistic values of impressions[0] and self._max_impressions, the difference is infinitesimally small and the user would never notice it.  If you want, I can investigate further and see if this is really necessary.

Reply:
Sorry that I forgot to reply to this.  Thank you for the detailed explanation, and that's quite careful!  I feel the need to be cautious here, and let's go with what it is.